### ISSUE TITLE
Add BeCheckPoint to HaikuArchives

### ISSUE THREAD

**Issue by richienyhus** [haikuports/haikuports#3055]
[becheckpoint.zip](https://github.com/haikuports/haikuports/files/2381834/becheckpoint.zip)

**Comment by korli** [comment]
Author: John Wiggins <john@ubermensch.net>

**Comment by scottmc** [comment]
License is now MIT, when moving this to HaikuArchives be sure to add an MIT license file as well.

**Comment by jwiggins** [comment]
Please refer to https://github.com/jwiggins/BeOS-Projects instead of that zip file.

**Comment by scottmc** [comment]
@jwiggins thanks for the update.  Would you like us to move any to HaikuArchives, or are you open to Pull Requests for changes?

**Comment by jwiggins** [comment]
My repository is mainly for historical purposes, so PRs will likely be ignored. I have no means to test changes made to that code.

**Comment by bach5000** [comment]
I'll try to add this to HaikuArchives as well. At least nothing seems broken here :sweat_smile:

**Comment by bach5000** [comment]
My repo for HaikuArchives is here: https://github.com/bach5000/BeCheckPoint.
Is this ready to merge?

**Comment by bach5000** [comment]
When I convert the BCP.rsrc file into an BCP.rdef file and change RSRCS to RDEFS in the Makefile, the terminal gives me an error:
```
rc: Error! -:68 syntax error, unexpected TYPECODE
/boot/system/develop/etc/makefile-engine:305: recipe for target 'objects.x86_64-cc7-release/BCP.rsrc' failed
```
I deleted the BCP.rsrc file after I replaced it with BCP.rdef, and I removed things in the RSRCS space in the Makefile. What am I doing wrong?

**Comment by scottmc** [comment]
Hmm.  Try reading through the rdef file and see if you spot anything.  Try using rc to convert the rdef back to rsrc and see if that will compile.

**Comment by fbrosson** [comment]
@scottmc, ~wouldn't it be better to:~ I was wondering if we could:
a) create an empty repo
b) fetch https://github.com/jwiggins/BeOS-Projects
c) merge all its commits that changed any file inside the BeCheckPoint subdirectory
d) identify the commits which might have changed files outside of the BeCheckPoint subdirectory and amend them in order to discard what's outside of the BeCheckPoint subdirectory.

Hmm, it seems this is going to be a lot of trouble to import just the BeCheckPoint subdirectory of the original repo. (What I was trying to do was preserve @jwiggins's authorship.)

@jwiggins, would you prefer that the HaikuPorts recipe for BeCheckPoint use tarballs from your repo instead of https://github.com/HaikuArchives/BeCheckPoint ? This would mean that you would get PRs from time to time...

**Comment by fbrosson** [comment]
Hmm, maybe @bach5000 can grab https://github.com/jwiggins/BeOS-Projects/commit/58b61bb4f4fca42b2c838674548bc4dab1f4b8d9 and merge it into an empty repo and then do a `git mv BeCheckPoint src` and a `git commit --amend` in order to populate a new fresh repo to have it transferred to https://github.com/HaikuArchives/BeCheckPoint.
(I'm suggesting this but I'm ok with anything else that satisfies @jwiggins.)

**Comment by scottmc** [comment]
I don't know that we ever want to make a recipe for this, at least not until it's been tested further and deemed safe to use on Haiku.  It's more like sample code that others might get ideas from.  It looks to have been be written in the R4 era, and may or may not be compatible with open tracker.  That said, if you see an easy path to clone just this and import it to Haiku archives we can try that.

**Comment by mmlr** [comment]
Due to launch_daemon this will probably not work right anymore. Instead of killing applications it would need to stop them via launch_daemon. Then again with the launch_daemon this functionality should be redone to not need to kill anything at all and instead hook into the proper place during startup. Also, how does this interact with ctrl-alt-delete?

Regarding extracting the sources out of a larger repository: This is exactly what git filter-branch with a subdirectory filter does.

**Comment by scottmc** [comment]
Once this task is completed we can add the launch-daemon info as a new issue/task. This might be an interesting exercise.  See if this can be done and whether the result would be useful.

As for this task, Bach did mention being rusty with git.  This might be a good test to try out git filter-branch and earn some foo points along the way.

**Comment by fbrosson** [comment]
Thanks @mmlr for suggesting `git filter-branch`. I gave it a glance but understood I would need much more time to understand how to use it correctly than what would take me to forge a new repo with the correct authorship :smile:
So here's my repo: https://github.com/fbrosson/BeCheckPoint

@bach5000, I've also preserved your authorship for the changes you did.

All options are open: We can continue with https://github.com/HaikuArchives/BeCheckPoint or remove it and do a new transfer from https://github.com/fbrosson/BeCheckPoint to https://github.com/HaikuArchives or try to use `git filter-branch`.

**Comment by bach5000** [comment]
@fbrosson I think your repo should do. Thanks for your help 👍 😄 

Do I still have to change the rsrc to rdef? Or do I just leave as it is (because it seems to work fine when I leave it) ? I did use the `rc` command as @scottmc had mentioned, but I still can't get it to work and keep getting the error mentioned above ☹️ 

And as for `git filter-branch`, I could try it out. It seems useful :)

**Comment by fbrosson** [comment]
@scottmc, the difference between my repo and the one that was transfered from @bach5000 is just that:
* My repo has the original authorship and timestamps from the author's commit, as we can see in https://github.com/fbrosson/BeCheckPoint/commits/master
* I did not have to [delete the `README.md`](https://github.com/HaikuArchives/BeCheckPoint/commit/71fdb439c1fac8ba14f60e983518c7ccfe152d8e) because I did not import it.

As said earlier, I'm ok with what is in https://github.com/HaikuArchives/BeCheckPoint right now. I created https://github.com/fbrosson/BeCheckPoint in case you would like redo a transfer.

@bach5000, I'm not familiar with the rsrc/rdef topic, so I would need to investigate... Let's hope another mentor can help you. (If you are stuck I can dig the topic later.)

**Comment by bach5000** [comment]
@fbrosson I just need to add the license file to the repo in https://github.com/HaikuArchives/BeCheckPoint (I made a PR: https://github.com/HaikuArchives/BeCheckPoint/pull/1) . Other than that, since I'm kind of stuck with the rsrc/rdef problem, I don't have much to do.

**Comment by scottmc** [comment]
Heres the article that explains rsrc vs rdef files and much more https://www.haiku-os.org/documents/dev/compile_them_resources/

I'll do a reimport on the repo in a couples hours here.

**Comment by scottmc** [comment]
Ok, repo moved and is now live.  Closing this issue as completed.  If you'd like to do more work with BeCheckPoint there is now a task to test it and report bugs and issues.  There's at least one typo i spotted in the readme that could use fixing as well.

**Comment by fbrosson** [comment]
Cool!
BTW, I've just deleted my temporary repo, so now everything will happen in https://github.com/HaikuArchives/BeCheckPoint.

**Comment by scottmc** [comment]
Seems there is no recipe for BeCheckPoint, and I think we should keep it that way unless the code is updated to better support Haiku.  Looks like LockWorkStation might be a better solution for users who might want such a program.  But BeCheckPoint is on HaikuDepot, brought in via a third party repo.
